### DISCUSSION TITLE
Roadmap suggestion: Suggest automation enhancement by adding sensors

### DISCUSSION THREAD

**Discussion by tsvi** [ITSpecialist111/ai_automation_suggester#16] [Ideas]
I suppose some automations can be improved in accuracy based on some additional inputs.

As an example, I have an automation that triggers based on the wifi BSSID my phone is connected to. Actually what I'm trying to achieve is to understand in which room I am. Obviously, the trigger won't work if I'm without my phone, or it might need some "massaging" to get it to be more exact.

If the AI could suggest some extra sensors that might give a better indication: add a camera, add a mmWave presence sensor, etc... that might be great (you might even monetize the affiliate purchase links)

**Comment by ITSpecialist111** [comment]
Thank you for your great ideas! I was wondering if we could create a small chatbot within HASS that would allow users to suggest the types of automations they would like to bring together. One challenge we face is associating each of the possible entities, which will require comprehensive intelligence across all entities. 

Additionally, I've noticed that there are very few comments on each of the entities, so I might need to explore the repository for more information. 

I'll take some time to think about how we can integrate these ideas. My initial thoughts involve using AI agents to gather information into a local repository, allowing AI to analyze the collected data. While this is a larger project, it is not impossible.

**Reply by Senbei123** [reply]
A small chatbot would be brilliant, because they you can give the ai modell feedback do the suggestion or even ask for details. 

Would it be an idea, to integrate your solution in the Assist interface?
![grafik](https://github.com/user-attachments/assets/29dfb9ae-dabd-4044-88b3-f113ee9ee381)

This could also be the place, where you show the response of the AI-Modell. (next to the perment storage in an entity.)

**Reply by ITSpecialist111** [reply]
That's not a bad idea! Let me see if if the assistant to call both - not tried that

**Reply by ITSpecialist111** [reply]
I can use the existing "assist" though to see the entities through the APIs, and with OpenAI (GPT-4), I can ask it to create the YAML for the automation based on those entities. Honestly, I don’t see much benefit in building another assistant unless I’m completely missing the point—it's been a long day, lol!

![image](https://github.com/user-attachments/assets/dc870e70-0d76-416d-a1b6-d9f4ffd86695)

**Reply by Senbei123** [reply]
Well, your integration with the power to collect entities + the very nice wrap up style TOGETHER with the possibilieties of the Home Assistant ASSIST to start a dialog about the outcome would be very very cool.
(Anyhow, I will promote your integration on my youtube channel in germany. ;))

**Reply by ITSpecialist111** [reply]
Thank you so much! I still have many ideas for enhancing the assistance capability. My initial thoughts remain focused on creating a dedicated dashboard that would display a list of daily suggestions. It would also include a chatbot for making changes, along with additional windows or cards to showcase the up-to-date YAML code for any sensors that need to be created, as well as the automation YAML. While reusing the built-in assist is a great idea, I would appreciate having more screen space to view everything.

Another idea I shared on Reddit involves building agents into Home Assistant (HASS). The concept is that HASS doesn't fully understand what it can see from the entities, so you could provide a knowledge base along with a set of custom instructions to add more context to the integrations and entities. This could include technical schematics, details, manuals, etc. The information could either be imported or linked to from an external source.

By the way, I would love to subscribe to your channel! Could you please send me the link?

DALL-E has tried to mock this up for me ;)

![image](https://github.com/user-attachments/assets/9a080ab3-c748-4513-bddc-2dafe2bb5dd7)

**Reply by Senbei123** [reply]
Very cool ideas @ITSpecialist111 ! This could be the start of something very powerfull! 

As a starting point, I wouldn't focus too much on the YAML part, because my experiance is, that those are most of the time wrong. 
But the general idee which is announced by the ai are often realy nice. And figgering out the correct entities for this use case and addapting the automation is then often realy easy.

For the "context" part, I have the idea, to give the ai modell also information how often and when the entity is triggerd. Eg. a motion sensor is trigger mostly in the early morning and evening. Next to this, in the same timeperiod the bathroom light goes on. The the ai could perhaps "think" that this could belong together and suggests an automation which triggers the light based on motion in a specific period of time.

Looking forward to get more ideas and development for this automation.

You find me at https://www.youtube.com/@tristanssmartesheim and the video will be go live on Monday at 5pm cet+1.

Tristan

**Comment by HL-Apprentice** [comment]
Hi @ITSpecialist111,
I love all the ideas coming through. I haven't seen this one yet, but I haven't combed through all the idea sections.
In addition to all the amazing ideas flowing here can you add a debugging "script"(lack of a better word) for the AI to monitor and possibly manage devices that seem to drop off randomly. see if there is any correlation with anything else going on?
I have different plugins just drop off. not knowing if it is a device, wifi, internet, or HA .
Something like, 
1. device integration takes a long time to recover from boot
2. integration looses connection randomly here are the logs, so it can be posted on the integration git( or automatically posted even.)
3. device drops of network randomly, (if available via HA show response times leading up to disconnection)
4.  device has been offline for N time, and the AI reloads the integration. then notifies of the device offline with logs.

having 263 devices and over 1900 entities is hard to keep track of and this would be amazing to keep HA in order even for us HA enthusiasts.

**Reply by ITSpecialist111** [reply]
I think this is a cool idea, and I really like it. I've been considering the possibility of generalizing this tool into a custom AI agent, allowing users to choose its functionalities. My goal is to have multiple agents running simultaneously to enhance capabilities. However, in light of MCP ( https://www.home-assistant.io/blog/2025/02/05/release-20252/#model-context-protocol ), I believe this might be a better approach. The drawback is that it would operate outside of HASS, but it would offer a more straightforward solution. What are your thoughts?

**Reply by HL-Apprentice** [reply]
I had to look up MCP. ;-)
HA has a bunch of other applications that work with it externally. Scripted, graphana, etc. as external machine/vm, etc.
If that is the method to connect my local LLM to HA with greater capabilities so be it.
I'm waiting for the day we can have a local Jarvis. LOL. 
I would love to have a local AI able to understand the HA environment, zigbee network, z-wave, bluetooth and wifi so it can see things that logging alone miss.  monitor the traffic for repeated requests, ping times, response times, etc.

**Reply by Hedda** [reply]
> I think this is a cool idea, and I really like it. I've been considering the possibility of generalizing this tool into a custom AI agent, allowing users to choose its functionalities. My goal is to have multiple agents running simultaneously to enhance capabilities. However, in light of MCP ( https://www.home-assistant.io/blog/2025/02/05/release-20252/#model-context-protocol ), I believe this might be a better approach. The drawback is that it would operate outside of HASS, but it would offer a more straightforward solution. What are your thoughts?

@ITSpecialist111 Be sure to check out this related MCP discussion under the instruction guide by allenporter here:

* https://gist.github.com/allenporter/b0e9946feb2ab60901c4f467ac1ba6f9

Also started a seperate discussion on MCP here on your repo:

* https://github.com/ITSpecialist111/ai_automation_suggester/discussions/52

**Reply by Hedda** [reply]
Another approach could maybe be "Fabric" AI conversion integration for Home Assistant?

- https://github.com/danielmiessler/fabric

***fabric** (from danielmiessler) is an open-source framework for augmenting humans using AI. It provides a modular framework for solving specific problems using a crowdsourced set of AI prompts that can be used anywhere.*

fabric intro video:

* [Network Chuck](https://www.youtube.com/watch?v=UbDyjIIGaxQ)
* [David Bombal](https://www.youtube.com/watch?v=vF-MQmVxnCs)
* [My Own Intro to the Tool](https://www.youtube.com/watch?v=wPEyyigh10g)
* [More Fabric YouTube Videos](https://www.youtube.com/results?search_query=fabric+ai)

**Reply by Hedda** [reply]
> I think this is a cool idea, and I really like it. I've been considering the possibility of generalizing this tool into a custom AI agent, allowing users to choose its functionalities. My goal is to have multiple agents running simultaneously to enhance capabilities. However, in light of MCP ( https://www.home-assistant.io/blog/2025/02/05/release-20252/#model-context-protocol ), I believe this might be a better approach. The drawback is that it would operate outside of HASS, but it would offer a more straightforward solution. What are your thoughts?

@ITSpecialist111 a specific use case for MCP server support could possibly be to use "codename goose" to generate YAML code example?

* https://github.com/block/goose

  - https://block.github.io/goose/

"_goose is your on-machine AI agent, capable of automating complex development tasks from start to finish. More than just code suggestions, goose can build entire projects from scratch, write and execute code, debug failures, orchestrate workflows, and interact with external APIs - autonomously_."

PS: Again, generic MCP server support discussion here -> https://github.com/ITSpecialist111/ai_automation_suggester/discussions/52

**Comment by ITSpecialist111** [comment]
Now that Google Gemini 2.5 and OpenAI 03 models can do 1m token windows. I've been allowing the models to look at ALL entities in my very large HASS setup. I'll eventually get around to MCP tomorrow depending on how my days go.

I've been trying different custom instructions to provide outputs like we've discussed above. Can you try and let me know your thoughts/improvements?

![image](https://github.com/user-attachments/assets/96405d33-0bd9-45db-9c9f-58d6406dbb1d)
